translate/translate.go

```diff
@@ -129,6 +134,85 @@ type Key struct {
 
 type informationElements map[Key]InformationElementEntry
 
+var reducedSizeErr error = errors.New("Unable to read reduced size encoding")
+
+// Helper method to read an unsigned reduced size field
+func reducedSizeReadUnsigned(bs []byte, maxBits int) (uint64, error) {
+	// Exit if `bs` has more bits than we can store
+	if len(bs)*8 > maxBits {
+		return 0, reducedSizeErr
+	}
+
+	switch len(bs) {
+	case 1:
+		return uint64(bs[0]), nil
+	case 2:
+		return uint64(binary.BigEndian.Uint16(bs)), nil
+	case 3:
+		return uint64(uint32(bs[0])<<16 + uint32(bs[1])<<8 + uint32(bs[2])), nil
+	case 4:
+		return uint64(binary.BigEndian.Uint32(bs)), nil
+	case 8:
+		return binary.BigEndian.Uint64(bs), nil
+	}
+	return 0, reducedSizeErr
+}
+
+// Helper method to read a signed reduced size field
+func reducedSizeReadSigned(bs []byte, maxBits int) (int64, error) {
```

Comment:
@matthew-dailey Yeah, this method is sticky for sure!  The tests already included cover these pretty well, but I added a couple of tests that touch `reducedSizeReadSigned` and `reducedSizeReadUnsigned` directly, and now test coverage is 100% for those methods.

<img width="486" alt="screenshot 2016-09-07 13 42 59" src="https://cloud.githubusercontent.com/assets/13948507/18326149/aab737b2-7501-11e6-81f3-a38708c90bbe.png">